app/controllers/manageBusinesses/add/IncomeSourceAddedController.scala

```diff
@@ -89,45 +109,49 @@ class IncomeSourceAddedController @Inject()(val authorisedFunctions: AuthorisedF
     }
   }
 
-  def handleSuccess(incomeSourceId: IncomeSourceId, incomeSourceType: IncomeSourceType, businessName: Option[String],
-                    showPreviousTaxYears: Boolean, isAgent: Boolean)(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
-    sessionService.getMongo(JourneyType(Add, incomeSourceType).toString).flatMap {
-      case Right(Some(sessionData)) =>
-        val oldAddIncomeSourceSessionData = sessionData.addIncomeSourceData.getOrElse(AddIncomeSourceData())
-        val updatedAddIncomeSourceSessionData = oldAddIncomeSourceSessionData.copy(journeyIsComplete = Some(true))
-        val uiJourneySessionData: UIJourneySessionData = sessionData.copy(addIncomeSourceData = Some(updatedAddIncomeSourceSessionData))
-        sessionService.setMongoData(uiJourneySessionData).flatMap { _ =>
-          incomeSourceType match {
-            case SelfEmployment =>
-              nextUpdatesService.getObligationsViewModel(incomeSourceId.value, showPreviousTaxYears) map { viewModel =>
-                Ok(obligationsView(businessName = businessName, sources = viewModel, isAgent = isAgent, incomeSourceType = SelfEmployment))
-              }
-            case _ => nextUpdatesService.getObligationsViewModel(incomeSourceId.value, showPreviousTaxYears) map { viewModel =>
-              Ok(obligationsView(viewModel, isAgent = isAgent, incomeSourceType = incomeSourceType))
-            }
-          }
-        }
-      case _ =>
-        val agentPrefix = if (isAgent) "[Agent]" else ""
-        Logger("application").error(agentPrefix + s"Unable to retrieve Mongo session data for $incomeSourceType")
-        Future.successful {
-          errorHandler(isAgent).showInternalServerError()
-        }
+  private def getReportingMethod(maybeAddIncomeSourceData: Option[AddIncomeSourceData]): ChosenReportingMethod = {
+    val (reportingMethodTaxYear1, reportingMethodTaxYear2) = (
+      maybeAddIncomeSourceData.flatMap(_.reportingMethodTaxYear1).orElse(None),
+      maybeAddIncomeSourceData.flatMap(_.reportingMethodTaxYear2).orElse(None)
+    )
+    (reportingMethodTaxYear1, reportingMethodTaxYear2) match {
+      case (Some("A"), Some("A")) | (None, Some("A")) => ChosenReportingMethod.Annual
+      case (Some("Q"), Some("Q")) | (None, Some("Q")) => ChosenReportingMethod.Quarterly
+      case (Some("A"), Some("Q")) | (Some("Q"), Some("A")) => ChosenReportingMethod.Hybrid
+      case (None, None) => ChosenReportingMethod.DefaultAnnual
+      case _ => ChosenReportingMethod.Unknown
```

Comment:
Defaults to annual if the reporting methods page is skipped, where it seems to only be skipped with the ITSA status for the current year is "A".

The unknown case should never happen (i.e. Some() for CY-1 and None for CY) since the user will need to choose reporting methods for both years where applicable.

Reply:
Need to confirm if Unknown case is fine as is, I believe it is and should never hit that case